components/centraldashboard/public/ajax_test_helper.js

```diff
@@ -26,3 +26,64 @@ export function mockRequest(component, response, respondWithError = false,
         component.addEventListener(responseEvent, () => resolve());
     });
 }
+
+/**
+ * Async sleep function which can pause execution flow or yield for next
+ * executions in the event-loop.
+ * @param {number} t Time in ms.
+ * @return {Promise<undefined>}
+ */
+export const sleep = (t) => new Promise((res) => setTimeout(res, t));
+
+
+/**
+ * Simply yields to the next tick, so that all Promised requests can do their
+ * thing.
+ */
+export async function yieldForRequests() {
+    await sleep(1); // Waits till next tick
+}
+
+/**
+ * Helper method that will mock a request directly on the iron-ajax element
+ * by overriding its `generateRequest()` method
+ * @param {HTMLElement} component Iron Ajax elements only
+ * @param {object} response
+ * @param {boolean=} respondWithError
+ */
+export function mockIronAjax(component, response, respondWithError = false) {
+    const {tagName, url} = component;
+    expect(tagName).toBe(
+        'IRON-AJAX', '[mockRequest] Element passed in was not iron-ajax'
+    );
+    // eslint-disable-next-line no-console
+    const finalEvent = `${respondWithError?'error':'response'}`;
+    component.generateRequest = async () => {
+        const eventPayload = createRespPayload(
+            response,
+            respondWithError,
+        );
+        if (respondWithError) {
+            component._setLastError(eventPayload.detail);
+        } else {
+            component._setLastResponse(response);
+        }
+        component.dispatchEvent(
+            new CustomEvent(finalEvent, eventPayload)
+        );
+        // eslint-disable-next-line no-console,max-len
+        console.log(`[mockIronAjax] ${url?url+' ':''}Dispatched ${finalEvent}`, eventPayload);
```

Comment:
Remove before submitting.

Reply:
I thought this would make sense to keep in for debugging. Though I can see how it would get annoying with a bigger test. Removing...